go/vt/vtgate/scatter_conn.go

```diff
@@ -260,6 +270,10 @@ func (stc *ScatterConn) ExecuteMultiShard(
 			mu.Lock()
 			defer mu.Unlock()
 
+			if innerqr != nil {
+				resultsObserver.observe(innerqr)
+			}
+
```

Comment:
Here it is given that innerqr will not be nil.
Just below, we append the result to the output result set as-sis without checking.